### ISSUE TITLE
auto learning rate finder fails AFTER optimization due to misconfiguration

### ISSUE THREAD

**Issue by noamzilo** [PyTorchLightning/pytorch-lightning#5487]
## 🐛 Bug
Optimizing learning rate using auto_lr_find=True fails **AFTER** a learning rate was found with error

`pytorch_lightning.utilities.exceptions.MisconfigurationException: When auto_lr_find is set to True, expects that `model` or `model.hparams` either has field `lr` or `learning_rate` that can overridden`

**AFTER** having waited for several minutes.

This can be caught **BEFORE** the long run to save time.

## Please reproduce using the BoringModel

I can just copy paste everything to here... the only lacking line is `self.lr = 999` under `model::__init__()`, if it is not present the error happens, otherwise it doesn't happen.

Reproducing this a single time takes me 15 minutes (due to the lr finding process), but I still wanted to report this because it seems important and I don't have time right now for a full reproduction on the template.

`
trainer = Trainer(auto_lr_find=True)
trainer.tune(model)
`

then for a while I see the progress bar

`Finding best initial lr:  27%|██▋       | 27/100 [02:43<07:37,  6.27s/it]`

and then the error after a lr has been found.

### Expected behavior

`pytorch_lightning.utilities.exceptions.MisconfigurationException: When auto_lr_find is set to True, expects that `model` or `model.hparams` either has field `lr` or `learning_rate` that can overridden`

should hit BEFORE the long calculation.

### Environment

 - PyTorch Version (e.g., 1.0): 1.7.1
 - OS (e.g., Linux): Windows 
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source): -
 - Python version: 3.7
 - CUDA/cuDNN version: 101
 - GPU models and configuration: single GPU NVIDIA quadro P4000
 - Any other relevant information:-

**Comment by rohitgr7** [comment]
good catch! Mind send a PR with a fix?

**Comment by noamzilo** [comment]
> 
> 
> good catch! Mind send a PR with a fix?

I actually never contributed to open source before :)

How does this work? I get assigned and there's a deadline?
is there a readme?

I want to do it!

**Comment by carmocca** [comment]
We have a basic guide [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md)!

There's no deadline! Any contribution is welcome :smile:

**Comment by noamzilo** [comment]
Can the same issue be done twice by some people w/o them knowing about it until the latter does a PR?

**Comment by carmocca** [comment]
Yes. For this reason, you might want to create the PR in draft mode while you are working on it, so others can know about it

https://github.blog/2019-02-14-introducing-draft-pull-requests/

**Comment by edenlightning** [comment]
Thanks for contributing @noamzilo! I have assigned this ticket to you, so other contributors know that this is already in progress. please let us know if you need any help!

**Comment by noamzilo** [comment]
How to link this post to the PR?
PR at:

https://github.com/PyTorchLightning/pytorch-lightning/issues/5487

currently my test doesn't pass due to lack of 

`pip install pytest-timeout` (from https://stackoverflow.com/questions/19527320/how-can-i-limit-the-maximum-running-time-for-a-unit-test) in the test server.

Please allow this library, or advise how I should limit test time.

Thanks :)
